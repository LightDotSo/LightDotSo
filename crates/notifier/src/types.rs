// Copyright 2023-2024 Light
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

use lazy_static::lazy_static;
use lightdotso_prisma::{ActivityEntity, ActivityOperation};
use serde_json::Value;
use std::collections::BTreeMap;
use strum::IntoEnumIterator;
use strum_macros::{Display, EnumIter, EnumString, IntoStaticStr};

// This is a list of all platforms
#[derive(Clone, Debug, EnumString, EnumIter, IntoStaticStr, Display)]
pub enum Platform {
    #[strum(serialize = "WEB")]
    Web,
}

// This is a list of all user operations
#[derive(Clone, Debug, EnumString, EnumIter, IntoStaticStr, Display)]
pub enum UserOnlyOperation {
    #[strum(serialize = "INVITE_CODE_ACCEPTED")]
    InviteCodeAccepted,
}

// Utility function to match the activity entity/operation combination
pub fn match_user_only_notification_with_activity(
    entity: &ActivityEntity,
    operation: &ActivityOperation,
    log: &Value,
) -> Option<UserOnlyOperation> {
    match (entity, operation) {
        (ActivityEntity::InviteCode, ActivityOperation::Update) => {
            // Check if the key of log is status is `USED`
            if log["status"].as_str() == Some("USED") {
                Some(UserOnlyOperation::InviteCodeAccepted)
            } else {
                None
            }
        }
        _ => None,
    }
}

// This is a list of all wallet operations
#[derive(Clone, Debug, EnumString, EnumIter, IntoStaticStr, Display)]
pub enum WalletOnlyOperation {
    #[strum(serialize = "USER_OPERATION_CREATED")]
    UserOperationCreated,
    #[strum(serialize = "USER_OPERATION_EXECUTED")]
    UserOperationExecuted,
    #[strum(serialize = "TRANSACTION_WITH_USER_OPERATION_EXECUTED")]
    TransactionWithUserOperationExecuted,
}

// Utility function to match the activity entity/operation combination
pub fn match_wallet_only_notification_with_activity(
    entity: &ActivityEntity,
    operation: &ActivityOperation,
    log: &Value,
) -> Option<WalletOnlyOperation> {
    match (entity, operation) {
        (ActivityEntity::UserOperation, ActivityOperation::Create) => {
            Some(WalletOnlyOperation::UserOperationCreated)
        }
        (ActivityEntity::UserOperation, ActivityOperation::Update) => {
            if log["status"].as_str() == Some("EXECUTED") ||
                log["status"].as_str() == Some("REVERTED")
            {
                Some(WalletOnlyOperation::TransactionWithUserOperationExecuted)
            } else {
                None
            }
        }
        (ActivityEntity::Transaction, ActivityOperation::Update) => None,
        _ => None,
    }
}

// Combination utility function to match the activity entity/operation combination
pub fn match_notification_with_activity(
    entity: &ActivityEntity,
    operation: &ActivityOperation,
    log: &Value,
) -> Option<Operation> {
    if let Some(res) = match_user_only_notification_with_activity(entity, operation, log) {
        return Some(Operation::UserOnly(res));
    }

    match_wallet_only_notification_with_activity(entity, operation, log).map(Operation::WalletOnly)
}

// This is a list of all operations
#[derive(Clone, Debug, Display)]
pub enum Operation {
    UserOnly(UserOnlyOperation),
    WalletOnly(WalletOnlyOperation),
}

lazy_static! {
    // This is a list of all operations
    pub static ref OPERATIONS: Vec<Operation> = {
        let mut operations = Vec::new();

        for operation in UserOnlyOperation::iter() {
            operations.push(Operation::UserOnly(operation));
        }

        for operation in WalletOnlyOperation::iter() {
            operations.push(Operation::WalletOnly(operation));
        }

        operations
    };
}

// For each operation, we define a notification type which is all combinations of operation and
// platform
#[derive(Clone, Debug)]
pub enum Notification {
    UserOnly(Platform, UserOnlyOperation),
    WalletOnly(Platform, WalletOnlyOperation),
}

// Implement Display for Notification
impl std::fmt::Display for Notification {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        match self {
            Notification::UserOnly(platform, operation) => {
                write!(f, "{}-{}", platform, operation)
            }
            Notification::WalletOnly(platform, operation) => {
                write!(f, "{}-{}", platform, operation)
            }
        }
    }
}

lazy_static! {
    // This is a list of all notification types
    pub static ref NOTIFICATION: Vec<Notification> = {
        let mut notification_types = Vec::new();

        for operation in UserOnlyOperation::iter() {
            for platform in Platform::iter() {
                notification_types.push(Notification::UserOnly(platform, operation.clone()));
            }
        }

        for operation in WalletOnlyOperation::iter() {
            for platform in Platform::iter() {
                notification_types.push(Notification::WalletOnly(platform, operation.clone()));
            }
        }

        notification_types
    };
}

lazy_static! {
    // This is a list of all user notification types
    pub static ref USER_NOTIFICATION_DEFAULT_ENABLED: BTreeMap<String, bool> = {
        let mut map = BTreeMap::new();

        for notif in NOTIFICATION.iter() {
            if let Notification::UserOnly(_, operation) = notif {
                let is_default_enabled = match operation {
                    UserOnlyOperation::InviteCodeAccepted => true,
                };
                map.insert(notif.to_string(), is_default_enabled);
            }
        }

        map
    };
}

lazy_static! {
    // This is a list of all wallet notification types
    pub static ref WALLET_NOTIFICATION_DEFAULT_ENABLED: BTreeMap<String, bool> = {
        let mut map = BTreeMap::new();

        for notif in NOTIFICATION.iter() {
            if let Notification::WalletOnly(_, operation) = notif {
                let is_default_enabled = match operation {
                    WalletOnlyOperation::UserOperationCreated => true,
                    WalletOnlyOperation::UserOperationExecuted => true,
                    WalletOnlyOperation::TransactionWithUserOperationExecuted => true,
                };
                map.insert(notif.to_string(), is_default_enabled);
            }
        }

        map
    };
}

lazy_static! {
    // This is a list of all notification default enabled
    pub static ref NOTIFICATION_DEFAULT_ENABLED: BTreeMap<String, bool> = {
        let mut map = BTreeMap::new();

        for (key, value) in &*USER_NOTIFICATION_DEFAULT_ENABLED {
            map.insert(key.clone(), *value);
        }

        for (key, value) in &*WALLET_NOTIFICATION_DEFAULT_ENABLED {
            map.insert(key.clone(), *value);
        }

        map
    };
}

lazy_static! {
    // This is a list of all user notification keys
    pub static ref USER_NOTIFICATION_KEYS: Vec<String> = {
        let mut user_notification_keys = Vec::new();

        for notif in NOTIFICATION.iter() {
            if let Notification::UserOnly(_, _) = notif {
                user_notification_keys.push(notif.to_string());
            }
        }

        user_notification_keys
    };
}

lazy_static! {
    // This is a list of all wallet notification keys
    pub static ref WALLET_NOTIFICATION_KEYS: Vec<String> = {
        let mut wallet_notification_keys = Vec::new();

        for notif in NOTIFICATION.iter() {
            if let Notification::WalletOnly(_, _) = notif {
                wallet_notification_keys.push(notif.to_string());
            }
        }

        wallet_notification_keys
    };
}

lazy_static! {
    // This is a list of all notification keys
    pub static ref NOTIFICATION_KEYS: Vec<String> = {
        let mut notification_keys = Vec::new();

        for notification in NOTIFICATION.iter() {
            notification_keys.push(notification.to_string());
        }

        notification_keys
    };
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_user_notification_keys() {
        for key in USER_NOTIFICATION_KEYS.iter() {
            println!("{}", key);
        }
        insta::assert_snapshot!("USER_NOTIFICATION_KEYS", format!("{:?}", *USER_NOTIFICATION_KEYS));
    }

    #[test]
    fn test_wallet_notification_keys() {
        for key in WALLET_NOTIFICATION_KEYS.iter() {
            println!("{}", key);
        }
        insta::assert_snapshot!(
            "WALLET_NOTIFICATION_KEYS",
            format!("{:?}", *WALLET_NOTIFICATION_KEYS)
        );
    }

    #[test]
    fn test_notification_keys() {
        for key in NOTIFICATION_KEYS.iter() {
            println!("{}", key);
        }
        insta::assert_snapshot!("NOTIFICATION_KEYS", format!("{:?}", *NOTIFICATION_KEYS));
    }

    #[test]
    fn test_operations() {
        for operation in OPERATIONS.iter() {
            println!("{}", operation);
        }
        insta::assert_snapshot!("OPERATIONS", format!("{:?}", *OPERATIONS));
    }

    #[test]
    fn test_notification() {
        for notification in NOTIFICATION.iter() {
            println!("{}", notification);
        }
        insta::assert_snapshot!("NOTIFICATION", format!("{:?}", *NOTIFICATION));
    }

    #[test]
    fn test_notification_default_enabled() {
        insta::assert_snapshot!(
            "NOTIFICATION_DEFAULT_ENABLED",
            format!("{:?}", *NOTIFICATION_DEFAULT_ENABLED)
        );
    }

    #[test]
    fn test_user_notification_default_enabled() {
        insta::assert_snapshot!(
            "USER_NOTIFICATION_DEFAULT_ENABLED",
            format!("{:?}", *USER_NOTIFICATION_DEFAULT_ENABLED)
        );
    }

    #[test]
    fn test_wallet_notification_default_enabled() {
        insta::assert_snapshot!(
            "WALLET_NOTIFICATION_DEFAULT_ENABLED",
            format!("{:?}", *WALLET_NOTIFICATION_DEFAULT_ENABLED)
        );
    }
}
