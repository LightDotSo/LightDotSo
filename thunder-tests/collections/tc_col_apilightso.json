{
  "_id": "90010c79-933e-468c-92a4-3ccac076fbb0",
  "colName": "api.light.so",
  "created": "2023-09-19T16:28:12.122Z",
  "sortNum": 5000,
  "folders": [],
  "requests": [
    {
      "_id": "809e5ace-0d9d-4285-a8b2-e3bda35ea70c",
      "colId": "90010c79-933e-468c-92a4-3ccac076fbb0",
      "containerId": "",
      "name": "https://api.light.so/check",
      "url": "https://api.light.so/check",
      "method": "GET",
      "sortNum": 10000,
      "created": "2023-09-19T16:28:12.124Z",
      "modified": "2023-09-19T16:28:12.124Z",
      "headers": [],
      "params": [],
      "tests": []
    },
    {
      "_id": "b00fe035-90b6-4984-8e18-2125d25db173",
      "colId": "90010c79-933e-468c-92a4-3ccac076fbb0",
      "containerId": "",
      "name": "wallet/list",
      "url": "https://api.light.so/wallet/list",
      "method": "GET",
      "sortNum": 20000,
      "created": "2023-10-12T04:32:59.438Z",
      "modified": "2023-10-12T04:33:11.914Z",
      "headers": [],
      "params": [],
      "tests": []
    }
  ]
}
