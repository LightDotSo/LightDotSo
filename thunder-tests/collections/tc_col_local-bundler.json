{
  "_id": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
  "colName": "local bundler",
  "created": "2023-09-25T17:16:43.618Z",
  "sortNum": 20000,
  "folders": [],
  "requests": [
    {
      "_id": "f5fc5074-e34d-4db7-9fb0-20b0cbc455c6",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_chainId local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 10000,
      "created": "2023-09-25T17:16:43.620Z",
      "modified": "2023-09-25T17:16:43.620Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_chainId\"\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "9bf81f8d-be96-47b2-bcf3-989235113f0e",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_supportedEntryPoints local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 20000,
      "created": "2023-09-25T17:19:01.660Z",
      "modified": "2023-09-25T17:19:28.615Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_supportedEntryPoints\"\n}",
        "form": []
      },
      "tests": []
    }
  ]
}
