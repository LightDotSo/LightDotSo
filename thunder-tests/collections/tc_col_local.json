{
  "_id": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
  "colName": "local",
  "created": "2023-09-25T17:16:43.618Z",
  "sortNum": 20000,
  "folders": [
    {
      "_id": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "api",
      "containerId": "",
      "created": "2023-10-12T04:34:09.664Z",
      "sortNum": 10000
    }
  ],
  "requests": [
    {
      "_id": "f5fc5074-e34d-4db7-9fb0-20b0cbc455c6",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_chainId local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 10000,
      "created": "2023-09-25T17:16:43.620Z",
      "modified": "2023-09-26T01:52:21.188Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_chainId\",\n  \"params\": [1]\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "9bf81f8d-be96-47b2-bcf3-989235113f0e",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_supportedEntryPoints local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 20000,
      "created": "2023-09-25T17:19:01.660Z",
      "modified": "2023-09-25T22:01:39.792Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_supportedEntryPoints\",\n  \"params\": [1]\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "36c23d43-9546-46bf-bb4f-58a7891f1457",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "paymaster_requestGasAndPaymasterAndData local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 15000,
      "created": "2023-09-25T18:24:20.613Z",
      "modified": "2023-10-30T15:52:09.636Z",
      "headers": [
        {
          "name": "accept",
          "value": "application/json"
        },
        {
          "name": "content-type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"paymaster_requestGasAndPaymasterAndData\",\n  \"params\": [\n    {\n      \"sender\": \"0x630ad0241296d73cac7c1ce9e798800051fa7fb0\",\n      \"nonce\": \"0x0\",\n      \"initCode\": \"0x0000000000756d3e6464f5efe7e413a0af1c7474183815c8b416eb7e5703c66bf1afc9e68a19f54087168e74b82eba920af3fde8dee8d15b000000000000000000000000000000000000000000000000000000000000108e\",\n      \"callData\": \"0x\",\n      \"signature\": \"0x000100000001000131a184eb40202a407819e4efe1313e8464c56ae6bb88ee91728134892f57a1df2519f8cd158ca4d60043fa37ed9da5e8748757367374a7c0ea745fdf364280c31c01\",\n      \"paymasterAndData\": \"0x\"\n    },\n    \"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789\",\n    10\n  ]\n}",
        "form": []
      },
      "tests": [
        {
          "type": "res-code",
          "custom": "",
          "action": "equal",
          "value": "200"
        },
        {
          "type": "res-body",
          "custom": "",
          "action": "contains",
          "value": "callGasLimit"
        }
      ]
    },
    {
      "_id": "80728a00-d34b-42dd-9d9d-e9eed8d6560e",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_estimateUserOperationGas local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 12500,
      "created": "2023-09-25T19:01:11.712Z",
      "modified": "2023-09-27T03:52:45.945Z",
      "headers": [
        {
          "name": "accept",
          "value": "application/json"
        },
        {
          "name": "content-type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_estimateUserOperationGas\",\n  \"params\": [\n    {\n      \"sender\": \"0x10dbbe70128929723c1b982e53c51653232e4ff2\",\n      \"nonce\": \"0x0\",\n      \"callData\": \"0x\",\n      \"initCode\": \"0x0000000000756d3e6464f5efe7e413a0af1c7474183815c8b7f285c774a1c925209bebaab24662b22e7cf32e2f7a412bfcb1bf52294b9ed60000000000000000000000000000000000000000000000000000000000000001\",\n      \"paymasterAndData\": \"0x\",\n      \"signature\": \"0x65a9427b82b04bac978a23357b45c2c03c16e119a52c3671a63cfa45811d22385f75f43307787376232c1420b5d75914529c308fd5e1fb0dd00277ab65d8206e1b01\",\n      \"maxFeePerGas\": \"0x08f0d188\",\n      \"maxPriorityFeePerGas\": \"0x08f0d188\",\n      \"preVerificationGas\": \"0xab90\",\n      \"verificationGasLimit\": \"0x989680\",\n      \"callGasLimit\": \"0x238c\"\n    },\n    \"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789\",\n    11155111\n  ]\n}",
        "form": []
      },
      "tests": [
        {
          "type": "res-body",
          "custom": "",
          "action": "contains",
          "value": "32500"
        }
      ]
    },
    {
      "_id": "d876726f-f608-422b-b66c-bc72a9005ccf",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_getUserOperationByHash local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 30000,
      "created": "2023-09-26T01:53:19.304Z",
      "modified": "2023-09-26T01:53:36.988Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_getUserOperationByHash\",\n  \"params\": [\n    \"0x520ed372ad45aab3fd3728a52ecbfb493e7e9affa388ff5b7810809f4313c29b\",\n    1\n  ]\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "6037bf43-80a4-4e00-869d-4d991fe6beb5",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_sendUserOperation local",
      "url": "https://rpc.light.so/11155111",
      "method": "POST",
      "sortNum": 40000,
      "created": "2023-09-27T01:28:43.929Z",
      "modified": "2023-09-27T01:28:43.929Z",
      "headers": [
        {
          "name": "accept",
          "value": "application/json"
        },
        {
          "name": "content-type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_sendUserOperation\",\n  \"params\": [\n    {\n      \"sender\": \"0x9aB4F3E162276BA0106739a729fdC29447D90FA6\",\n      \"nonce\": \"0x0\",\n      \"initCode\": \"0x0000000000756d3e6464f5efe7e413a0af1c7474183815c8b416eb7e5703c66bf1afc9e68a19f54087168e74b82eba920af3fde8dee8d15b0000000000000000000000000000000000000000000000000000000000000674\",\n      \"callData\": \"0x\",\n      \"signature\": \"0x2f90b195460d34fe1183dedc74775f31028aef4ac030b15e319bae46ebcd985273195cef8d1854e18661222a982685d8805b4f2d35a918479198ccdcc3864c5b1b01\",\n      \"paymasterAndData\": \"0x000000000018d32df916ff115a25fbefc70baf8b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000650cdd6d1407a016f30168736d0441798019f2447e6eeebbb09324ce8282a3d548f8826a1b19b8829fa9b76d971561be8b97f498e70e5ab4441d964739d644deb7eb09011b\",\n      \"preVerificationGas\": \"0xab84\",\n      \"verificationGasLimit\": \"0x63faa\",\n      \"callGasLimit\": \"0x238c\",\n      \"maxFeePerGas\": \"0x2a6\",\n      \"maxPriorityFeePerGas\": \"0x2a6\"\n    },\n    \"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789\"\n  ]\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "24769435-2ab4-460f-b6a8-3d55929672aa",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "gas_requestGasEstimation local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 11250,
      "created": "2023-09-28T15:33:43.927Z",
      "modified": "2023-10-27T15:21:42.992Z",
      "headers": [
        {
          "name": "accept",
          "value": "application/json"
        },
        {
          "name": "content-type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"gas_requestGasEstimation\",\n  \"params\": [\n    1\n  ]\n}",
        "form": []
      },
      "tests": [
        {
          "type": "res-code",
          "custom": "",
          "action": "equal",
          "value": "200"
        }
      ]
    },
    {
      "_id": "39bc9158-31c6-4ee2-8b86-a431fce84227",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "",
      "name": "eth_getUserOperationReceipt local",
      "url": "http://localhost:3000",
      "method": "POST",
      "sortNum": 35000,
      "created": "2023-09-29T19:34:56.657Z",
      "modified": "2023-09-29T19:35:12.344Z",
      "headers": [
        {
          "name": "Content-Type",
          "value": "application/json"
        }
      ],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"id\": 1,\n  \"jsonrpc\": \"2.0\",\n  \"method\": \"eth_getUserOperationReceipt\",\n  \"params\": [\n    \"0x520ed372ad45aab3fd3728a52ecbfb493e7e9affa388ff5b7810809f4313c29b\",\n    1\n  ]\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "835e534d-c378-4d68-97de-62586e98299b",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "configuration/get",
      "url": "http://localhost:3000/v1/configuration/get?address=0x0d4428aA9Bc2e854477CcCC9135883947EAEb120",
      "method": "GET",
      "sortNum": 50000,
      "created": "2023-10-12T04:34:19.773Z",
      "modified": "2023-10-12T19:51:49.185Z",
      "headers": [],
      "params": [
        {
          "name": "address",
          "value": "0x0d4428aA9Bc2e854477CcCC9135883947EAEb120",
          "isPath": false
        }
      ],
      "tests": []
    },
    {
      "_id": "b6754c6e-415a-4eec-b5df-09e436c21a64",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "wallet/list",
      "url": "http://localhost:3000/v1/wallet/list",
      "method": "GET",
      "sortNum": 60000,
      "created": "2023-10-12T04:34:39.909Z",
      "modified": "2023-10-12T19:52:06.278Z",
      "headers": [],
      "params": [],
      "tests": []
    },
    {
      "_id": "e06af7cb-41e7-4578-aaf2-6275e9980309",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "wallet/get",
      "url": "http://localhost:3000/v1/wallet/get?address=0x0d4428aA9Bc2e854477CcCC9135883947EAEb120",
      "method": "GET",
      "sortNum": 55000,
      "created": "2023-10-12T04:35:05.507Z",
      "modified": "2023-10-12T19:52:02.673Z",
      "headers": [],
      "params": [
        {
          "name": "address",
          "value": "0x0d4428aA9Bc2e854477CcCC9135883947EAEb120",
          "isPath": false
        }
      ],
      "tests": []
    },
    {
      "_id": "bf5fd786-5577-43eb-b689-91eebdf6194d",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "configuration/list",
      "url": "http://localhost:3000/v1/configuration/list",
      "method": "GET",
      "sortNum": 52500,
      "created": "2023-10-12T04:41:06.558Z",
      "modified": "2023-10-12T19:51:51.105Z",
      "headers": [],
      "params": [],
      "tests": []
    },
    {
      "_id": "5792f306-79ea-40eb-ae85-47fe91db5195",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "wallet/create",
      "url": "http://localhost:3000/v1/wallet/create",
      "method": "POST",
      "sortNum": 53750,
      "created": "2023-10-12T05:05:54.598Z",
      "modified": "2023-10-12T05:10:02.591Z",
      "headers": [],
      "params": [],
      "body": {
        "type": "json",
        "raw": "{\n  \"owners\": [\n    {\n      \"address\": \"0x4fd9D0eE6D6564E80A9Ee00c0163fC952d0A45Ed\",\n      \"weight\": 1\n    }\n  ],\n  \"salt\": \"0x0000000000000000000000000000000000000000000000000000000000000006\",\n  \"threshold\": 1\n}",
        "form": []
      },
      "tests": []
    },
    {
      "_id": "5b1591cd-8718-4713-a4e4-8d9e9c41e267",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "admin/wallet/list",
      "url": "http://localhost:3000/admin/v1/user_operation/list?address=0x6A10913f726978491ffd0e9e8A4a6f0FCeB1cD80",
      "method": "GET",
      "sortNum": 70000,
      "created": "2023-10-12T20:21:25.546Z",
      "modified": "2023-10-30T18:24:39.486Z",
      "headers": [
        {
          "name": "Authorization",
          "value": "Bearer dev"
        }
      ],
      "params": [
        {
          "name": "address",
          "value": "0x6A10913f726978491ffd0e9e8A4a6f0FCeB1cD80",
          "isPath": false
        }
      ],
      "tests": []
    },
    {
      "_id": "f532a7d2-1c66-4d2c-9a40-bec278c07ec4",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "user_operation/list",
      "url": "http://localhost:3000/v1/user_operation/list",
      "method": "GET",
      "sortNum": 65000,
      "created": "2023-10-19T19:23:12.301Z",
      "modified": "2023-10-19T19:23:25.630Z",
      "headers": [],
      "params": [],
      "tests": []
    },
    {
      "_id": "18448cbc-42e1-4117-bc83-25c6ed8428ce",
      "colId": "a73c26e2-ef46-4e67-9789-a8c630d3d989",
      "containerId": "22a50e81-1b69-4f0a-a9b6-6cff82c6a067",
      "name": "user_operation/signature",
      "url": "http://localhost:3000/admin/v1/user_operation/signature?user_operation_hash=0x281976129b40530f25e59a310a6cb8d58831dd9b9a5f1d583f5e330c8aa6b0e0",
      "method": "GET",
      "sortNum": 67500,
      "created": "2023-11-08T23:51:58.495Z",
      "modified": "2023-11-09T07:00:12.972Z",
      "headers": [
        {
          "name": "Authorization",
          "value": "Bearer dev"
        }
      ],
      "params": [
        {
          "name": "user_operation_hash",
          "value": "0x281976129b40530f25e59a310a6cb8d58831dd9b9a5f1d583f5e330c8aa6b0e0",
          "isPath": false
        }
      ],
      "tests": []
    }
  ]
}
