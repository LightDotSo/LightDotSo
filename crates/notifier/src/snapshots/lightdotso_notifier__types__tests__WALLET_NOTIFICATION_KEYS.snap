---
source: crates/notifier/src/types.rs
expression: "format!(\"{:?}\", * WALLET_NOTIFICATION_KEYS)"
---
["WEB-USER_OPERATION_CREATED", "WEB-USER_OPERATION_EXECUTED", "WEB-TRANSACTION_WITH_USER_OPERATION_EXECUTED"]
